### PR TITLE
Datetime mergeasof tolerance

### LINKED ISSUE #28098: merge_asof() tolerance cannot be datetime.timedelta
#### Code Sample
```
import pandas as pd
import numpy as np
from datetime import timedelta

left = pd.DataFrame(
    list(zip([0, 5, 10, 15, 20, 25], [0, 1, 2, 3, 4, 5])),
    columns=["time", "left"],
)
left["time"] = pd.to_timedelta(left["time"], "ms")

right = pd.DataFrame(
    list(zip([0, 3, 9, 12, 15, 18], [0, 1, 2, 3, 4, 5])),
    columns=["time", "right"],
)
right["time"] = pd.to_timedelta(right["time"], "ms")

expected = pd.DataFrame(
    list(
        zip(
            [0, 5, 10, 15, 20, 25],
            [0, 1, 2, 3, 4, 5],
            [0, np.nan, 2, 4, np.nan, np.nan],
        )
    ),
    columns=["time", "left", "right"],
)
expected["time"] = pd.to_timedelta(expected["time"], "ms")
print(expected)

# this throws MergeError: incompatible tolerance <class 'datetime.timedelta'>, must be compat with type dtype('<m8[ns]')
result = pd.merge_asof(left, right, on="time", tolerance=timedelta(microseconds=1000), direction="nearest")

# this works!
# result = pd.merge_asof(left, right, on="time", tolerance=pd.Timedelta("1ms"), direction="nearest")

print(result)
```
#### Problem description
I can use pd.Timedelta as tolerance, but not datetime.timedelta. 

#### Expected Output
No error

#### Output of ``pd.show_versions()``

<details>
INSTALLED VERSIONS
------------------
commit           : None
python           : 3.7.3.final.0
python-bits      : 64
OS               : Linux
OS-release       : 4.19.0-5-amd64
machine          : x86_64
processor        : 
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 0.25.0+222.gdef01cf7b
numpy            : 1.17.0
pytz             : 2019.2
dateutil         : 2.8.0
pip              : 19.2.2
setuptools       : 41.2.0
Cython           : 0.29.13
pytest           : 5.1.1
hypothesis       : 4.33.1
sphinx           : 1.8.5
blosc            : 1.8.1
feather          : None
xlsxwriter       : 1.1.9
lxml.etree       : 4.4.1
html5lib         : 1.0.1
pymysql          : None
psycopg2         : None
jinja2           : 2.10.1
IPython          : 7.7.0
pandas_datareader: None
bs4              : 4.8.0
bottleneck       : 1.2.1
fastparquet      : 0.3.2
gcsfs            : None
lxml.etree       : 4.4.1
matplotlib       : 3.1.1
numexpr          : 2.7.0
odfpy            : None
openpyxl         : 2.6.3
pandas_gbq       : None
pyarrow          : 0.14.1
pytables         : None
s3fs             : 0.3.3
scipy            : 1.3.1
sqlalchemy       : 1.3.7
tables           : 3.5.2
xarray           : 0.12.3
xlrd             : 1.2.0
xlwt             : 1.3.0
xlsxwriter       : 1.1.9
</details>

### PR DESCRIPTION
**Pull request by ianzur**
- [x] closes #28098
- [x] tests 1 / 1
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/core/reshape/merge.py [hunk 1]
```diff
@@ -1705,7 +1706,11 @@ def flip(xs):
             left_values = left_values.view("i8")
             right_values = right_values.view("i8")
             if tolerance is not None:
-                tolerance = tolerance.value
+                try:
```
Comment (jreback) [inline pandas/core/reshape/merge.py]: this is not a good way
rather just wrap it in Timedelta then leave the original

**Review by jreback — CHANGES_REQUESTED** [review]
comments

File: doc/source/whatsnew/v0.25.1.rst [hunk 2]
```diff
@@ -99,6 +99,7 @@ Reshaping
 - Bug in :meth:`DataFrame.crosstab` when ``margins`` set to ``True`` and ``normalize`` is not ``False``, an error is raised. (:issue:`27500`)
 - :meth:`DataFrame.join` now suppresses the ``FutureWarning`` when the sort parameter is specified (:issue:`21952`)
 - Bug in :meth:`DataFrame.join` raising with readonly arrays (:issue:`27943`)
+- Bug :meth:`merge_asof` could not use datetime.timedelta for `tolerance` kwarg (:issue:`28098`)
```
Comment (mroeschke) [inline doc/source/whatsnew/v0.25.1.rst]: * ````Bug in :func:`merge_asof` ````
* ````:class:`datetime.timedelta` ````
* ```` ``tolerance`` ````

File: pandas/core/reshape/merge.py [hunk 3]
```diff
@@ -1705,6 +1706,8 @@ def flip(xs):
             left_values = left_values.view("i8")
             right_values = right_values.view("i8")
             if tolerance is not None:
+                if not isinstance(self.tolerance, Timedelta):
```
Comment (mroeschke) [inline pandas/core/reshape/merge.py]: Don't think we need this `if` statement. (it's okay to pass a `Timedelta` into `Timedelta` I think).

File: doc/source/whatsnew/v0.25.1.rst (hunk 2 above)
Comment (jreback) [reply doc/source/whatsnew/v0.25.1.rst]: move this to 0.25.2

**Review by jreback — CHANGES_REQUESTED** [review]

File: doc/source/whatsnew/v0.25.1.rst (hunk 2 above)
Comment (jreback) [reply doc/source/whatsnew/v0.25.1.rst]: sorry, move to 1.0

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by alimcmaster1** [comment]
@ianzur - this is great! Mind updating as per @jreback comments looks like this is almost there

**Comment by ianzur** [comment]
Thanks for the reminder! I got distracted with school.

**Comment by alimcmaster1** [comment]
Does this need a test? Can we now get rid of the xfail for test_tolerance in test_merge_asof?

**Comment by ianzur** [comment]
@alimcmaster1 There is already a test in place. The test is not set to xfail.

https://github.com/pandas-dev/pandas/blob/8de8b081a77a484cd0e4e73dd88abe9de99bfd75/pandas/tests/reshape/merge/test_merge_asof.py#L593-L605

**Comment by alimcmaster1** [comment]
@ianzur - LGTM now thanks! Yep see that now (wasn’t showing up earlier)

**Review by WillAyd — APPROVED** [review]
lgtm though @mroeschke or @jbrockmendel might have some thoughts

**Comment by mroeschke** [comment]
Thanks @ianzur.

**Pull request merged**